Add visualize_as_svg method to Instance + Solution classes

rahularya50 opened the pull request:
Tests added and all pass. Also checked visually that the SVGs look good, [example](https://user-images.githubusercontent.com/18374604/162605722-fc074d08-b8d2-4eb4-b624-8db0a84e7e5f.svg).

What do you think of adding a light grid background to the image? To make the lattice clear, especially for small inputs. Probably there should be a threshold in the config for when the grid is too dense and so should be omitted.

I also added a fixture in the tests so I didn't have to keep copying the valid instance/solutions into each test.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #10
* __->__ #9
* #8
* #1

rahularya50 committed:
Add visualize_as_svg method to Instance + Solution classes

[ghstack-poisoned]

rahularya50 committed:
Update on "Add visualize_as_svg method to Instance + Solution classes"




[ghstack-poisoned]

rahularya50 committed:
Update on "Add visualize_as_svg method to Instance + Solution classes"




[ghstack-poisoned]

rahularya50 commented on python/instance.py:
```diff
@@ -1,10 +1,14 @@
+from point import Point
+from svg import SVGGraphic
+
+if TYPE_CHECKING:
```
This is a meme in the stdlib:
```python
# Constant that's True when type checking, but False here.
TYPE_CHECKING = False
```
To avoid a circular import in the next diff.

tylerhou replied:
Not a huge fan of circular dependencies, but I think it's fine for now. I would prefer a trait-based system where Instance and Solution impl SvgSerialization or something, but that is too much work. (Same with regular text serialization.)

tylerhou commented on python/instance_test.py:
```diff
@@ -145,9 +149,11 @@ def test_serialize(self):
             ],
         )
```
I prefer making copying the instances vs. a setUp method; it's easier to read the test by itself and verify that it's correct.

https://testing.googleblog.com/2019/12/testing-on-toilet-tests-too-dry-make.html

tylerhou commented on python/solution_test.py:
```diff
@@ -145,66 +148,78 @@ def test_penalty(self):
             ],
             instance=instance,
```
Same here.

tylerhou commented:
I wish GitHub let me give approval but require that comments be acknowledged/resolved...

rahularya50 committed:
Update on "Add visualize_as_svg method to Instance + Solution classes"

Tests added and all pass. Also checked visually that the SVGs look good, [example](https://user-images.githubusercontent.com/18374604/162605722-fc074d08-b8d2-4eb4-b624-8db0a84e7e5f.svg).

What do you think of adding a light grid background to the image? To make the lattice clear, especially for small inputs. Probably there should be a threshold in the config for when the grid is too dense and so should be omitted.

I also added a fixture in the tests so I didn't have to keep copying the valid instance/solutions into each test.




[ghstack-poisoned]

rahularya50 commented:
sorry for the diff noise, IDE autoformatter doesn't like your code style
